packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -548,6 +562,56 @@ describe('paybackLoan success', () => {
     expect(rawtx.vin[0].txid).toStrictEqual(utxo.txid)
     expect(rawtx.vin[0].vout).toStrictEqual(utxo.vout)
   })
+
+  // fixes on master-5772b6c
+  it('should paybackLoan with DUSD', async () => {
+    const dusdBobAddr = await bob.generateAddress()
+    await bob.rpc.loan.takeLoan({
+      vaultId: bobVaultId,
+      to: dusdBobAddr,
+      amounts: '19@DUSD'
+    })
+    await bob.generate(1)
+
+    const vaultBefore = await bob.container.call('getvault', [bobVaultId])
+    expect(vaultBefore).toStrictEqual({
+      vaultId: bobVaultId,
+      loanSchemeId: 'scheme',
+      ownerAddress: bobVaultAddr,
+      state: 'active',
+      collateralAmounts: ['10000.00000000@DFI', '1.00000000@BTC'],
+      loanAmounts: ['19.00001084@DUSD', '40.00004566@TSLA'],
+      interestAmounts: ['0.00001084@DUSD', '0.00004566@TSLA'],
+      collateralValue: 15000,
+      loanValue: 99.00010216,
+      interestValue: 0.00010216,
+      informativeRatio: 15151.49951639,
+      collateralRatio: 15151
+    })
+
+    await bob.rpc.loan.paybackLoan({
+      vaultId: bobVaultId,
+      amounts: '19@DUSD',
+      from: dusdBobAddr
+    })
+    await bob.generate(1)
+
+    const vaultAfter = await bob.container.call('getvault', [bobVaultId])
+    expect(vaultAfter).toStrictEqual({
+      vaultId: bobVaultId,
+      loanSchemeId: 'scheme',
+      ownerAddress: bobVaultAddr,
+      state: 'active',
+      collateralAmounts: ['10000.00000000@DFI', '1.00000000@BTC'],
+      loanAmounts: ['0.00001084@DUSD', '40.00006849@TSLA'],
+      interestAmounts: ['0.00000000@DUSD', '0.00006849@TSLA'],
```

Comment:
Just curious, 0 DUSD interest is because the interest of 0.00001084 is too small? Shouldn't here will continue accrue dust among our interest?

Reply:
nope.. the amount paid the interest first

Reply:
I remember previously the behavior of `interestAmount` is a projected amount if no more payment in current block? Is that fixed? I understand the previous interest are paid, but current remaining 0.00001084 will continue incur interest right

Reply:
> I remember previously the behavior of interestAmount is a projected amount if no more payment in current block? Is that fixed?
yep.. the interest block height re-allocated calculation is fixed.. no projected amount.

> I understand the previous interest are paid, but current remaining 0.00001084 will continue incur interest right
yup the remaining `0.00001084` should incur the interest.. its worth to test on this dust amount

Reply:
> yup the remaining 0.00001084 should incur the interest.. its worth to test on this dust amount
@ivan-zynesis 
bug :lady_beetle:  :sweat: